ghcide/src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -66,7 +67,9 @@ plugin :: Plugin c
 plugin = codeActionPluginWithRules rules codeAction <> Plugin mempty setHandlersCodeLens
 
 rules :: Rules ()
-rules = rulePackageExports
+rules = do
+  rulePackageExports
+  getAnnotatedParsedSourceRule
```

Comment:
Why is this put in the rules in the CodeAction plugin? Why do code action plugins define any additional rules to add to the graph?

Reply:
Plugins have the ability to define new rules. Are you asking why this rule is not part of the main set of rules?

Reply:
It's more I expect whoever defines this type to also be the one to put it on to the set of rules - are IDE.Plugin.CodeAction and IDE.GHC.ExactPrint the same "thing"?

Reply:
No, they're not, so I guess it belongs with the core set of rules